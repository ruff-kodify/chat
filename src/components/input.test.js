import React from 'react';
import Input from './input';

function createInput(props) {
  const wrapper = mount(<Input { ...props } />);
  const input = wrapper.find('[data-test="input"]');
  const button = wrapper.find('[data-test="button"]');
  return {
    wrapper,
    input,
    button
  };
}

test('it should render properly', () => {
  const wrapper = shallow(<Input />);
  expect(wrapper).toMatchSnapshot();
});

test('it should call onSend if user clicks on the button', () => {
  const value = 'hello world';
  const props = {
    onSend: jest.fn()
  };
  const { input, button } = createInput(props);
  input.simulate('change', { target: { value } });
  button.simulate('click');
  expect(props.onSend).toHaveBeenCalledTimes(1);
  expect(props.onSend).toHaveBeenCalledWith(value);
});

test('it should call onSend if the user presses enter', () => {
  const value = 'hello world';
  const props = {
    onSend: jest.fn()
  };
  const { input } = createInput(props);
  input.simulate('change', { target: { value: 'hello world' } });
  input.simulate('keyup', { keyCode: 13 });
  expect(props.onSend).toHaveBeenCalledTimes(1);
  expect(props.onSend).toHaveBeenCalledWith(value);
});

test('it should call onType if the user is writing some text', () => {
  const props = {
    onType: jest.fn()
  };
  const { input } = createInput(props);
  input.simulate('keyup', { target: { value: 'h' } });
  input.simulate('keyup', { target: { value: 'he' } });
  input.simulate('keyup', { target: { value: 'hel' } });
  input.simulate('keyup', { target: { value: 'hell' } });
  input.simulate('keyup', { target: { value: 'hello' } });

  expect(props.onType).toHaveBeenCalledTimes(5);
  expect(props.onType).toHaveBeenCalledWith('h');
  expect(props.onType).toHaveBeenLastCalledWith('hello');
});
