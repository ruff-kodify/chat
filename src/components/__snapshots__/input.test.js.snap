// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render properly 1`] = `
<div
  className="input"
>
  <input
    data-test="input"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value=""
  />
  <button
    data-test="button"
    onClick={[Function]}
  >
    Send
  </button>
</div>
`;
